pkgs/development/python-modules/langgraph-cli/default.nix

```diff
@@ -0,0 +1,65 @@
+{
+  lib,
+  buildPythonApplication,
+  click,
+  fetchFromGitHub,
+  nix-update-script,
+  poetry-core,
+  pytest-asyncio,
+  pytestCheckHook,
+  pythonOlder,
+}:
+
+buildPythonApplication rec {
+  pname = "langgraph";
+  version = "0.1.49";
+  pyproject = true;
+
+  disabled = pythonOlder "3.10";
+
+  src = fetchFromGitHub {
+    owner = "langchain-ai";
+    repo = "langgraph";
+    rev = "refs/tags/cli==${version}";
+    hash = "sha256-yphXboJA/TLGIPggb2Cvsxn1+WUSYMzC0wPHft3TGvo=";
+  };
+
+  sourceRoot = "${src.name}/libs/cli";
+
+  build-system = [ poetry-core ];
+
+  dependencies = [ click ];
+
+  nativeCheckInputs = [
+    pytest-asyncio
+    pytestCheckHook
+  ];
+
+  pytestFlagsArray = [ "tests/unit_tests" ];
+
+  disabledTests = [
+    # Flaky tests that generate a Docker configuration then compare to exact text
+    "test_config_to_docker_simple"
+    "test_config_to_docker_pipconfig"
+    "test_config_to_compose_env_vars"
+    "test_config_to_compose_env_file"
+    "test_config_to_compose_end_to_end"
+    "test_config_to_compose_simple_config"
+    "test_config_to_compose_watch"
+  ];
+
+  passthru.updateScript = nix-update-script {
+    extraArgs = [
+      "--version-regex"
+      "cli==(.*)"
+    ];
+  };
+
+  meta = with lib; {
```

Comment:
I think you can easily remove `with lib;` and also add `meta.mainProgram`.

Reply:
mainProgram: Done

`with lib;` is how `nix-init` does it.

Reply:
It's unfortunate that `nix-init` is barely maintained at the moment.

There are numerous [PRs](https://github.com/NixOS/nixpkgs/pulls?q=is%3Apr+%22remove+with+lib%22+is%3Aclosed+sort%3Aupdated-desc+) aimed at removing `with lib;` when placed in front of a set.

Technically, what you've done is fine, and I completely understand if you choose not to make the change. However, it's important to keep in mind that `nixpkgs` is a dynamic and constantly evolving repository. Practices that were acceptable yesterday might change tomorrow.  As a maintainer, I sometimes find it challenging to grasp the necessity of certain changes. For instance, `with lib;` is commonly avoided in front of a set because it merges the set's contents with those of `lib`, complicating static analysis and creating hurdles for newcomers. While using it in front of the `meta` set might not cause significant issues, adhering to best coding practices is still valuable. For this reason, I prefer to see it removed, in alignment with the efforts of many other maintainers.

Reply:
Fixed across the board. Thank you :)

Reply:
Done!